### PR TITLE
CRIMAPP-512 Savings summary page

### PR DESCRIPTION
**Pull request by hiboabd**
## Description of change
Adds savings summary page displaying any savings 
Adds change and remove actions 
Change actions routes user to the savings details page to edit details of saving 
Remove action takes user to confirm screen to confirm they would like to remove the saving. 
When saving is deleted user is routed back to the saving summary page if there are savings or the saving type page if there are no more savings 
Also adds question to user to add another savings account. If yes is selected they are routed to the savings type page, if no is selected they are routed to the premium bonds page (not yet implemented)

## Link to relevant ticket
[CRIMAPP-512](https://dsdmoj.atlassian.net/browse/CRIMAPP-512)
[CRIMAPP-513](https://dsdmoj.atlassian.net/browse/CRIMAPP-513) - to add confirm remove screen

## Notes for reviewer
This is the first of two PRs. The following CRIMAPP-512 PR will fix the display of the savings on the page to match design (currently just a list of the savings attributes are being displayed)

### After changes:

**Saving summary page** 
Screenshot 2024-02-23 at 14 30 12

**Validation error if user clicks save and continue without selecting option** 
Screenshot 2024-02-23 at 14 30 25

**Deletion of savings account routes to savings summary page if there are other savings** 
Screenshot 2024-02-23 at 14 33 08
Screenshot 2024-02-23 at 14 35 53

**Deletion of a savings account routes to savings type page if there are no other savings** 
Screenshot 2024-02-23 at 14 38 29
Screenshot 2024-02-23 at 13 23 22

## How to manually test the feature
Navigate to `steps/capital/which_savings_does_client_have`
Add a saving type and go to next screen
Fill in details of saving to add saving
Now you should be on the saving summary screen 
You should see the saving account you just added

Test the following:
- That you can click the change link and go to the saving detail page and update the saving 
- That you can click remove and are taken to the confirm removal screen. Then test you can click no and go back to the savings summary. Also test you can click yes and delete the saving
- That when you delete a saving and there are other savings, you are taken to the savings summary page
- That when you delete a saving and there are no other savings, you are taken to the premium bonds page (currently not implemented)
- On the saving summary page, click save and continue without answering the question of whether you want to add another saving account. Check you are shown a validation error 

Can also test that when you click save and continue whilst filling in saving details and navigate to the `steps/capital/clients_savings`, you are shown an incomplete tag for the saving with partial information filled

[CRIMAPP-512]: https://dsdmoj.atlassian.net/browse/CRIMAPP-512?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMAPP-513]: https://dsdmoj.atlassian.net/browse/CRIMAPP-513?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -223,7 +223,7 @@
     t.boolean "passporting_benefit"
     t.string "benefit_type"
     t.string "has_benefit_evidence"
-    t.index ["crime_application_id"], name: "index_people_on_crime_application_id", unique: true
+    t.index ["type", "crime_application_id"], name: "index_people_on_type_and_crime_application_id", unique: true
```
Comment (hiboabd) [inline db/schema.rb:226]: I keep getting this diff when I migrate, is anyone else getting this?

Comment (timpeat) [reply db/schema.rb:226]: @hiboabd delete your dev database. I think you have this from running migrations in our partner spike.

File: app/controllers/steps/capital/savings_controller.rb [hunk 2]
```diff
@@ -13,6 +13,21 @@ def update
...
+          redirect_to edit_steps_capital_savings_summary_path, success: t('.success_flash')
+        else
+          # If this was the last remaining record, redirect to the saving type page
+          redirect_to edit_steps_capital_saving_type_path, success: t('.success_flash')
+        end
+      end
+
+      def confirm_destroy
+        @saving = helpers.present(saving_record)
+      end
```
Comment (mseedat-moj) [inline app/controllers/steps/capital/savings_controller.rb:29]: Not sure what is happening here a comment may be helpful.

**Review by mseedat-moj — COMMENTED** [review]
Looks great - some questions from me

**Comment by ripan** [comment]
LGTM 👀 

Minor UI issue.  We are just showing values in "Show bank details" section. Should we include keys as well?
Screenshot 2024-02-27 at 12 24 55

**Comment by hiboabd** [comment]
> LGTM 👀
> 
> Minor UI issue. We are just showing values in "Show bank details" section. Should we include keys as well? Screenshot 2024-02-27 at 12 24 55

Thanks, i don't think keys are needed as the second pr will fix the display of the savings

File: app/controllers/steps/capital/savings_controller.rb (hunk 2 above)
Comment (hiboabd) [reply app/controllers/steps/capital/savings_controller.rb:29]: I think it's just setting the saving that will be displayed on the confirm destroy page. We have a savings presenter that will present some of the fields that need formatting but I haven't done that in this PR so I've removed for now and will add back and add a comment for the next PR.

File: app/controllers/steps/capital/savings_controller.rb [hunk 3]
```diff
@@ -13,6 +13,21 @@ def update
...
+      def destroy
+        saving_record.destroy
+
+        if savings.reload.any?
+          redirect_to edit_steps_capital_savings_summary_path, success: t('.success_flash')
+        else
+          # If this was the last remaining record, redirect to the saving type page
+          redirect_to edit_steps_capital_saving_type_path, success: t('.success_flash')
+        end
+      end
```
[the remark this answers is not in the document]
Comment (hiboabd) [reply app/controllers/steps/capital/savings_controller.rb:25]: Discussed with wider team. Agreed to put a ticket in the backlog to return to refactoring this across the codebase once the means journey is complete

**Review by mseedat-moj — APPROVED** [review]
Excellent

**Pull request merged**